lib/model.js

```diff
@@ -748,51 +747,67 @@ module.exports = (function() {
     });
   };
 
-  //right now, the caller (has-many-double-linked) is in charge of the where clause
-  Model.prototype.findAllJoin = function(joinTableName, options, queryOptions) {
-    options = optClone(options || {});
-    options.attributes = options.attributes || [this.QueryInterface.quoteTable(this.name) + '.*'];
-    // whereCollection is used for non-primary key updates
-    this.options.whereCollection = options.where || null;
-
-    return this.QueryInterface.select(this, [[this.getTableName(options), this.name], joinTableName], options, Utils._.defaults(queryOptions, { transaction: (options || {}).transaction }));
-  };
-
  /**
-  * Search for a single instance. This applies LIMIT 1, so the listener will always be called with a single instance.
+  * Search for a single instance by its primary key. This applies LIMIT 1, so the listener will always be called with a single instance.
   *
-  * @param  {Object|Number}             [options] A hash of options to describe the scope of the search, or a number to search by id.
+  * @param  {Number|String|Buffer}      [options] A hash of options to describe the scope of the search, or a number to search by id.
+  * @param  {Object}             '      [options]
   * @param  {Transaction}               [options.transaction]
-  * @param  {Object}                    [queryOptions]
   *
-  * @see {Model#findAll}           for an explanation of options and queryOptions
+  * @see {Model#findAll}           for an explanation of options
   * @return {Promise<Instance>}
-  * @alias find
+  * @alias findByPrimary
   */
-  Model.prototype.findOne = function(param, queryOptions) {
-    // For sanity findOne will never return if no options are passed
+  Model.prototype.findById = function(param, options) {
+    // For sanity findById will never return if no options are passed
     if ([null, undefined].indexOf(param) !== -1) {
       return Promise.resolve(null);
     }
 
-    var options = {};
+    options = options || {};
+    delete options.where;
```

Comment:
Using `options.where = undefined;` would be nicer (if nothing checks for the options keys - but _.defaults does not).